packages/expo-audio/android/src/main/java/expo/modules/audio/service/AudioRecordingService.kt

```diff
@@ -0,0 +1,201 @@
+
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.app.Service
+import android.content.Context
+import android.content.Intent
+import android.content.pm.ServiceInfo
+import android.os.Binder
+import android.os.Build
+import android.os.IBinder
+import android.util.Log
+import androidx.core.app.NotificationCompat
+import expo.modules.audio.AudioRecorder
+import java.lang.ref.WeakReference
+
+private const val TAG = "AudioRecordingService"
+
+class AudioRecordingService : Service() {
+  private val binder = AudioRecordingBinder()
+  private var activeRecorders = mutableSetOf<WeakReference<AudioRecorder>>()
+  private var notificationId = NOTIFICATION_ID
+
+  inner class AudioRecordingBinder : Binder() {
+    fun getService(): AudioRecordingService = this@AudioRecordingService
+  }
+
+  override fun onCreate() {
+    super.onCreate()
+    Log.d(TAG, "Service onCreate()")
+    instance = this
+    createNotificationChannelIfNeeded()
+
+    synchronized(pendingRecorders) {
+      pendingRecorders.forEach { recorder ->
+        registerRecorder(recorder)
+      }
+      pendingRecorders.clear()
+    }
+  }
+
+  override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+    when (intent?.action) {
+      ACTION_START_RECORDING -> {
+        if (activeRecorders.isNotEmpty()) {
+          startForegroundWithNotification()
+        }
+      }
+      ACTION_STOP_RECORDING -> {
+        stopRecordingAndService()
+      }
+    }
+    return START_NOT_STICKY
+  }
+
+  private fun createNotificationChannelIfNeeded() {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+      val notificationManager = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+      if (notificationManager.getNotificationChannel(CHANNEL_ID) == null) {
+        val channel = NotificationChannel(
+          CHANNEL_ID,
+          "Audio Recording",
+          NotificationManager.IMPORTANCE_LOW
+        ).apply {
+          description = "Shows when audio is being recorded in the background"
+        }
+        notificationManager.createNotificationChannel(channel)
+      }
+    }
+  }
+
+  private fun buildNotification(): Notification {
+    val stopIntent = Intent(this, AudioRecordingService::class.java).apply {
+      action = ACTION_STOP_RECORDING
+    }
+    val stopPendingIntent = PendingIntent.getService(
+      this,
+      0,
+      stopIntent,
+      PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
+    )
+
+    val contentIntent = packageManager.getLaunchIntentForPackage(packageName)?.let { appIntent ->
+      PendingIntent.getActivity(
+        this,
+        0,
+        appIntent,
+        PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
+      )
+    }
+
+    return NotificationCompat.Builder(this, CHANNEL_ID)
+      .setContentTitle("Recording audio")
```

Comment:
Should we make these strings configurable somehow? Feels like they'd very soon be a thing people would like to change/update.